src/banded/BandedMatrix.jl

```diff
@@ -43,12 +43,14 @@ const BandedSubBandedMatrix{T} =
 
 returns an uninitialized `n`×`m` banded matrix of type `T` with bandwidths `(l,u)`.
 """
-BandedMatrix{T}(::UndefInitializer, n::Integer, m::Integer, a::Integer, b::Integer) where {T<:BlasFloat} =
-    _BandedMatrix(Matrix{T}(undef,max(0,b+a+1),m), n, a, b)
-BandedMatrix{T}(::UndefInitializer, n::Integer, m::Integer, a::Integer, b::Integer)  where {T<:Number} =
-    _BandedMatrix(zeros(T,max(0,b+a+1),m),n,a,b)
+BandedMatrix{T, C}(::UndefInitializer, n::Integer, m::Integer, a::Integer, b::Integer) where {T<:BlasFloat, C<:AbstractMatrix{T}} =
+    _BandedMatrix(C(undef,max(0,b+a+1),m), n, a, b)
+BandedMatrix{T, C}(::UndefInitializer, n::Integer, m::Integer, a::Integer, b::Integer) where {T, C<:AbstractMatrix{T}} =
+    _BandedMatrix(C(undef,max(0,b+a+1),m),n,a,b)
+BandedMatrix{T, C}(::UndefInitializer, n::Integer, m::Integer, a::Integer, b::Integer)  where {T<:Number, C<:AbstractMatrix{T}} =
+    _BandedMatrix(fill!(similar(C, max(0,b+a+1),m), zero(T)),n,a,b)
```

Comment:
It's a bit ugly, but it seems to be the easiest way to create an initialized matrix (eg not undef) with the right type C